api/write.go

```diff
@@ -194,6 +197,8 @@ x:
 // Close finishes outstanding write operations,
 // stop background routines and closes all channels
 func (w *WriteAPIImpl) Close() {
+	w.closingMu.Lock()
```

Comment:
Since this is an exported type, users could (and may already) call `WriteAPIImpl{}` without instantiating this new mutex, so it could default to `nil`. Does doing a `nil` check make sense here before acquiring the lock?

Reply:
As there is no nil check for other fields, it is not necessary.